jgrapht-core/src/main/java/org/jgrapht/graph/MaskEdgeSet.java

```diff
@@ -77,8 +74,16 @@ public MaskEdgeSet(
      */
     @Override public boolean contains(Object o)
     {
-        return this.edgeSet.contains(o)
-            && !this.mask.isEdgeMasked(TypeUtil.uncheckedCast(o, edgeTypeDecl));
+        // Type system shenanigans to check whether o is an edge
+        try {
```

Comment:
Without it, this would throw a ClassCastException if contains() is called on something that isn't an E, which I think users would find surprising.

Unfortunately, I couldn't use the more legible "o instanceof E" because the generic type E is erased. I don't fully understand why this approach doesn't have the same problem, but that's Java for you.